### PR TITLE
document custom fields on cart discounts

### LINKED ISSUE #1616: document custom fields on cart discounts
Add support for our backend feature [CTP-849](https://github.com/sphereio/dev.commercetools.com/pull/746/files)

### PR DESCRIPTION
**Pull request by acbeni**
Closes #1616

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/CartDiscount.java [hunk 1]
```diff
@@ -119,6 +120,16 @@
     @HasUpdateAction(value = "changeTarget",className = "ChangeTarget")
     CartDiscountTarget getTarget();
 
+
+    /**
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/CartDiscount.java]: Nice that you added the javadoc here 👍

File: commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/CartDiscountDraft.java [hunk 2]
```diff
@@ -47,6 +48,8 @@
 
     CartDiscountValue getValue();
 
+    @Nullable
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/CartDiscountDraft.java]: How about adding/copying the javadoc here too? 😉

File: commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/commands/updateactions/SetCustomField.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package io.sphere.sdk.cartdiscounts.commands.updateactions;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.sphere.sdk.cartdiscounts.CartDiscount;
+import io.sphere.sdk.json.SphereJsonUtils;
+import io.sphere.sdk.types.customupdateactions.SetCustomFieldBase;
+
+/**
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/commands/updateactions/SetCustomField.java]: Nice javadoc 👍

File: commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/commands/updateactions/SetCustomType.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package io.sphere.sdk.cartdiscounts.commands.updateactions;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.sphere.sdk.cartdiscounts.CartDiscount;
+import io.sphere.sdk.types.customupdateactions.SetCustomTypeBase;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+/**
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/cartdiscounts/commands/updateactions/SetCustomType.java]: Nice javadoc 👍

**Comment by acbeni** [comment]
good catch @katmatt 👍 
i forgot to add javadoc to the draft, i will do it before merging.

**Pull request merged**